### PR TITLE
SF-2165 - Added design system formatting utilities

### PR DESCRIPTION
**Pull request by smbecker**

### CODE REVIEW

File: src/sdk/formatting/formatRange.test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+describe('formatRange', () => {
+	it('can format ranges', () => {
+		expect(formatRange('a', 'b')).to.eql('a\xa0-\xa0b');
+		expect(formatRange('100', '200')).to.eql('100\xa0-\xa0200');
+	});
+
+	it('can format with custom separator', () => {
+		expect(formatRange('a', 'b', {separator: '-'})).to.eql('a-b');
+	});
+});
```
Comment (rklind) [inline src/sdk/formatting/formatRange.test.ts]: should we add a couple of tests for start being null, end being null, and both being null?

File: src/sdk/formatting/formatPercent.test.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	it('can format negative percents', () => {
+		expect(formatPercent(-0.1)).to.eql('-10%');
+		expect(formatPercent(-0.2)).to.eql('-20%');
+		expect(formatPercent(-1.2)).to.eql('-120%');
+	});
+
+	it('should restrict decimals to 1 place', () => {
+		expect(formatPercent(0.2426345)).to.eql('24.3%');
+	});
+});
```
Comment (rklind) [inline src/sdk/formatting/formatPercent.test.ts]: should there be a test for a non number?  should there be a test for 0?

File: src/sdk/formatting/formatBytes.test.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	it('can format large negative tb', () => {
+		expect(formatBytes(-Math.pow(2, 40) * 1024)).to.eql('-1\xa0PB');
+		expect(formatBytes(-Math.pow(2, 40) * 1.213 * 1024)).to.eql('-1.2\xa0PB');
+		expect(formatBytes(-Math.pow(2, 40) * 9.81 * 1024)).to.eql('-9.8\xa0PB');
+	});
+
+	it('can format large yb', () => {
+		expect(formatBytes(Math.pow(2, 80) * 1024)).to.eql('1,024\xa0YB');
+	});
+});
```
Comment (rklind) [inline src/sdk/formatting/formatBytes.test.ts]: should there be a test for 0 and for a non number?

File: src/sdk/formatting/trimPreceedingZero.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const trimPreceedingZero = (value: string): string => {
```
[the remark this answers is not in the document]
Comment (smbecker) [reply src/sdk/formatting/trimPreceedingZero.ts]: Not sure what you mean? The only instance in the confluence document that mentions trimming preceding zeros is when dealing with time. In most cases, the time format automatically trims the preceding zero. However, I found that using `hour12:false` option removes the `AM/PM` suffix, but it also causes the preceding zero to not be removed. This function is to work-around that.

File: src/sdk/formatting/formatUnit.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const formatUnit = (value: string, unit?: string, separator?: string): string => {
```
[the remark this answers is not in the document]
Comment (smbecker) [reply src/sdk/formatting/formatUnit.ts]: We implicitly test this already with the `formatByte` and `formatNumber` tests.

File: src/sdk/formatting/RelativeDate.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	}
+	return date;
+};
+
+const second = 1e3;
+const minute = 6e4;
+const hour = 36e5;
+const day = 864e5;
+
+export class RelativeDate {
```
[the remark this answers is not in the document]
Comment (smbecker) [reply src/sdk/formatting/RelativeDate.ts]: We implicitly test this with the `formatDate` tests.

File: src/sdk/formatting/formatBytes.test.ts (hunk 3 above)
Comment (smbecker) [reply src/sdk/formatting/formatBytes.test.ts]: Done

File: src/sdk/formatting/formatPercent.test.ts (hunk 2 above)
Comment (smbecker) [reply src/sdk/formatting/formatPercent.test.ts]: Done

File: src/sdk/formatting/formatRange.test.ts (hunk 1 above)
Comment (smbecker) [reply src/sdk/formatting/formatRange.test.ts]: Done

File: src/sdk/formatting/convertDate.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+export type ConvertibleDate = Date | number | string;
+
+export const convertDate = (value: ConvertibleDate): Date => {
```
[the remark this answers is not in the document]
Comment (smbecker) [reply src/sdk/formatting/convertDate.ts]: Done
